src/main/java/com/mitchej123/hodgepodge/mixins/late/morpheus/MixinMorpheusWakePlayers.java

```diff
@@ -0,0 +1,27 @@
+package com.mitchej123.hodgepodge.mixins.late.morpheus;
+
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+import net.quetzi.morpheus.SleepChecker;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(value = SleepChecker.class, remap = false)
+public class MixinMorpheusWakePlayers {
+
+    @Inject(
+            method = "advanceToMorning",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/world/WorldProvider;resetRainAndThunder()V",
+                    shift = At.Shift.BEFORE),
+            cancellable = true)
+    public void hodgepodge$fixWakePlayers(World world, CallbackInfo c) {
+        if (!(world instanceof WorldServer worldServer)) return;
+        worldServer.wakeAllPlayers();
+        c.cancel();
```

Comment:
You should not cancel the method all the time, what if another mod injects code after your handler ? It will never run the other code. Does it really matter is the rain and thunder method is called twice anyways ?